tests/testthat/test-densities.R

```diff
@@ -0,0 +1,57 @@
+context("densities")
+
+## This one is the easiest:
+test_that("dpois agrees", {
```

Comment:
I think we should add an edge case test to all of these:
n = 0; p/lambda = 0; p = 1 are the ones I can think of (and perhaps combinations of these)

`dpois(1, 0, log = TRUE) = -Inf` is the kind of thing that motivated `rexp`

Reply:
good shout - added the cases I could think of here, can add more now or later as we find them